api/v1beta1/ocicluster_webhook.go

```diff
@@ -125,6 +133,12 @@ func (c *OCICluster) validate() field.ErrorList {
 			field.Invalid(field.NewPath("spec", "ociResourceIdentifier"), c.Spec.OCIResourceIdentifier, "field is required"))
 	}
 
+	if len(c.Spec.Region) > 0 && !validRegion(c.Spec.Region) {
```

Comment:
I cannot gather this, but we should call validate of network spec in update as well I think.